src/org/opensolaris/opengrok/configuration/Project.java

```diff
@@ -224,6 +249,29 @@ public void addGroup(Group group) {
     }
 
     /**
+     * Fill the project with the current configuration where the applicable
+     * project property has a default value.
+     *
+     * @param cfg configuration with default values if applicable
+     */
+    final public void completeWithDefaults(Configuration cfg) {
+        Configuration defaultCfg = new Configuration();
+        /**
+         * Choosing strategy for properties (tabSize here):
+         * <pre>
+         * this       cfg        defaultCfg
+         * ===========================================
+         *  |5|        4             0
```

Comment:
Not sure the table helps to understand; at least for me it makes things confusing. Maybe some extended description would help ? Or add another column with resulting value ?

Reply:
actually the table was somewhat wrong

I corrected the table and added a brief description of what's happening. The point was to fill the project properities only when it is set to default.